controllers/common/fetch.go

```diff
@@ -115,57 +119,69 @@ func FetchSubscription(c client.Client, name, namespace string, packageName ...s
 
 // FetchClusterServiceVersion fetch the ClusterServiceVersion from the subscription
 func FetchClusterServiceVersion(c client.Client, sub *olmv1alpha1.Subscription) (*olmv1alpha1.ClusterServiceVersion, error) {
-	// Check the ClusterServiceVersion status in the subscription
-	if sub.Status.CurrentCSV == "" {
-		klog.Warningf("The ClusterServiceVersion for Subscription %s is not ready. Will check it again", sub.Name)
-		return nil, nil
-	}
-
-	csvName := sub.Status.CurrentCSV
+	csvName := ""
 	csvNamespace := sub.Namespace
 
-	if sub.Status.Install == nil || sub.Status.InstallPlanRef.Name == "" {
+	if sub.Status.InstallPlanRef != nil {
+		ip, err := FetchInstallPlan(c, sub.Status.InstallPlanRef.Name, sub.Status.InstallPlanRef.Namespace)
+		if err != nil {
+			return nil, err
+		}
+		for _, step := range ip.Status.Plan {
+			if step.Resource.Kind == "ClusterServiceVersion" && strings.HasPrefix(step.Resource.Name, sub.Name) {
+				if step.Status == olmv1alpha1.StepStatusPresent {
```

Comment:
@DanielXLee When I fresh install, I found when the CSV is created, the status of the installplan is `olmv1alpha1.StepStatusCreated` instead of `olmv1alpha1.StepStatusPresent`.